src/vm_pages.rs

```diff
@@ -331,6 +402,41 @@ impl VmRegionList {
     }
 }
 
+// A reference to a `VmRegion` that is in the process of being updated. The referenced region
+// is converted back into its previous region type if this object is dropped before `finish()`
+// is called.
```

Comment:
nit: is "Reserved" a bit too generic as a name? It _is_ reserved as it is not accessible, but it doesn't describe really the state is in. Naming is hard, so won't insist on this.

Reply:
Yeah, fair. I renamed this to `VmRegionUpdater`, and changed the temporary region name to `Updating`. WDYT?